compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -85,11 +85,10 @@ pub(crate) fn new_item_kind(kind: DefKind) -> ItemKind {
         | DefKind::Field
         | DefKind::LifetimeParam
         | DefKind::Impl { .. }
-        | DefKind::Ctor(_, _)
         | DefKind::GlobalAsm => {
             unreachable!("Not a valid item kind: {kind:?}");
         }
-        DefKind::Closure | DefKind::AssocFn | DefKind::Fn => ItemKind::Fn,
+        DefKind::Ctor(_, _) | DefKind::Closure | DefKind::AssocFn | DefKind::Fn => ItemKind::Fn,
```

Comment:
This is not right for `CtorKind::Const`.

Reply:
That's fair. I'm assuming that will still have a body, right? In that case, should we classify them as `Const` instead or would you create a new ItemKind?

Reply:
Yes, they should be consts.

Reply:
Done